File: js/discount-applier.js
class DiscountApplier {
  constructor (notifier) {
    this.notifier = notifier
  }

  applyV1 (discount, users) {
    // Bug! Should be `let i = 0`
    for (let i = 1; i < users.length; i++) {
      const message = `You've got a new discount of ${discount}`
      const user = users[i]
      this.notifier.notify(user, message)
    }
  }

  applyV2 (discount, users) {
    for (let i = 0; i < users.length; i++) {
      const message = `You've got a new discount of ${discount}`
      // Bug! Should be `users[i]`
      const user = users[0]
      this.notifier.notify(user, message)
    }
  }
}

module.exports = DiscountApplier

File: python/discount_applier.py
class DiscountApplier:
    def __init__(self, notifier):
        self.notifier = notifier

    def apply_v1(self, discount, users):
        # Bug! Should be `range(0, len(users))`
        for i in range(1, len(users)):
            message = f"You've got a new discount of {discount}"
            user = users[i]
            self.notifier.notify(user, message)

    def apply_v2(self, discount, users):
        for i in range(0, len(users)):
            message = f"You've got a new discount of {discount}"
            # Bug! Should be `users[i]`
            user = users[0]
            self.notifier.notify(user, message)

